aerospike_helpers/cdt_ctx.py

```diff
@@ -300,3 +300,29 @@ def cdt_ctx_map_key_create(key: any, order: int = 0) -> _cdt_ctx:
         :class:`~aerospike_helpers.cdt_ctx._cdt_ctx`
     """
     return _cdt_ctx(id=aerospike.CDT_CTX_MAP_KEY_CREATE, value=key, extra_args={CDT_CTX_ORDER_KEY: order})
+
+def cdt_ctx_all_children() -> _cdt_ctx:
+    """
+    At the current context, causes a query to return a list of all the children
+    of the current item. For a map, this will recurse into the map elements.
+    For a list, this will include all the children in the list.
+
+    Returns:
+        :class:`~aerospike_helpers.cdt_ctx._cdt_ctx`
+    """
+    return _cdt_ctx(id=aerospike._CDT_CTX_EXP)
+
+def cdt_ctx_all_children_with_filter(expression: "TypeExpression") -> _cdt_ctx:
+    """
+    All children of the current level will be selected, and then the filter expression
+    is applied to each item in turn.  Items that cause the expression to evaluate to true will be added to the
+    list of items returned in a query for this level.  Items that cause the expression to evaluate to false
+    will be filtered out
+
+    Args:
+        expression: Compiled expression. This expression must return a boolean.
+
+    Returns:
+        :class:`~aerospike_helpers.cdt_ctx._cdt_ctx`
+    """
+    return _cdt_ctx(id=aerospike._CDT_CTX_EXP, extra_args={aerospike._CDT_CTX_ALL_CHILDREN_WITH_FILTER_EXPR_KEY: expression})
```

Comment:
I think its worth considering that the string literal "filter_expr" might be better to add to this object.
In other cases, like `aerospike._CDT_CTX_EXP`, the variables represent an integer value, and the variable name is able to describe what the value does better than a magic number.
In this case however, the underlying string value "filter_expr" is sufficient in describing the value, and in my opinion, `_CDT_CTX_ALL_CHILDREN_WITH_FILTER_EXPR_KEY` is overly complicated and very wordy.  
It is not a big deal and you could defintely leave it as is, but I think using the "filter_exp" string literal is more concise and descriptive in this scenario.
You could also rename it something simpler, like `_CDT_CTX_FILTER_EXPR_KEY`.

Reply:
Originally I was using `"filter_exp"` here, but then I decided to change it because I was running into issues when debugging my code only to find that the string in `aerospike_helpers` didn't match the one in `aerospike`'s C code. (the map key for expressions in the Python client's C code isn't consistent; sometimes it's "expr", other times "expressions", etc.) If I define it once in the C code and expose it as a constant, then I wouldn't need to change the string in the Python code as well.

> You could also rename it something simpler, like _CDT_CTX_FILTER_EXPR_KEY.

I agree with this.